File: app/src/main/java/com/main/climbingdiary/controller/button/AppFloatingActionButton.java
package com.main.climbingdiary.controller.button;

import android.annotation.SuppressLint;
import android.support.design.widget.FloatingActionButton;
import android.view.View;

import com.main.climbingdiary.R;
import com.main.climbingdiary.activities.MainActivity;
import com.main.climbingdiary.common.preferences.AppPreferenceManager;
import com.main.climbingdiary.dialog.DialogFactory;
import com.main.climbingdiary.models.Tabs;

public class AppFloatingActionButton implements View.OnClickListener {

    @SuppressLint("StaticFieldLeak")
    private static FloatingActionButton floatingActionButtonAdd;
    @SuppressLint("StaticFieldLeak")
    private static FloatingActionButton floatingActionButtonLocate;

    public AppFloatingActionButton() {
        floatingActionButtonAdd = MainActivity.getMainActivity().findViewById(R.id.floating_action_btn_add);
        floatingActionButtonLocate = MainActivity.getMainActivity().findViewById(R.id.floating_action_btn_locate);
        floatingActionButtonAdd.setOnClickListener(this);
        floatingActionButtonLocate.setOnClickListener(this);
    }

    @Override
    public void onClick(View v) {

        DialogFactory.openAddRouteDialog(AppPreferenceManager.getSelectedTabsTitle());
    }

    public static void show() {
        hide();
        if (AppPreferenceManager.getSelectedTabsTitle() == Tabs.MAP) {
            floatingActionButtonLocate.show();
        } else {
            floatingActionButtonAdd.show();
        }
    }

    public static void hide() {
        floatingActionButtonAdd.hide();
        floatingActionButtonLocate.hide();
    }
}

File: app/src/main/java/com/main/climbingdiary/controller/button/AppFloatingActionButton.kt
package com.main.climbingdiary.controller.button

import android.annotation.SuppressLint
import android.view.View
import androidx.appcompat.app.AppCompatActivity
import com.google.android.material.floatingactionbutton.FloatingActionButton
import com.main.climbingdiary.R
import com.main.climbingdiary.activities.MainActivity
import com.main.climbingdiary.common.preferences.AppPreferenceManager.getSelectedTabsTitle
import com.main.climbingdiary.dialog.DialogFactory

@SuppressLint("StaticFieldLeak")
object AppFloatingActionButton : View.OnClickListener {

    val activity: AppCompatActivity by lazy { MainActivity.getMainActivity() }
    private val floatingActionButtonAdd: FloatingActionButton = activity
        .findViewById(R.id.floating_action_btn_add)

    init {
        floatingActionButtonAdd.setOnClickListener(this)
    }

    fun show() {
        hide()
        floatingActionButtonAdd.show()
    }

    fun hide() {
        floatingActionButtonAdd.hide()
    }


    override fun onClick(v: View?) {
        getSelectedTabsTitle().let { DialogFactory.openAddRouteDialog(it) }
    }
}